packages/react-select/src/Select.tsx

```diff
@@ -478,8 +481,21 @@ function getNextFocusedOption<
   Option,
   IsMulti extends boolean,
   Group extends GroupBase<Option>
->(state: State<Option, IsMulti, Group>, options: Options<Option>) {
+>(
+  state: State<Option, IsMulti, Group>,
+  options: Options<Option>,
+  props?: Props<Option, IsMulti, Group>
+) {
   const { focusedOption: lastFocusedOption } = state;
+
+  if (props) {
+    const { alwaysFocusFirstMenuOption } = props;
+
+    if (alwaysFocusFirstMenuOption) {
+      return options[0];
+    }
+  }
+
   return lastFocusedOption && options.indexOf(lastFocusedOption) > -1
     ? lastFocusedOption
     : options[0];
```

Comment:
Why pass the whole props object? Maybe passing only the relevant prop here would be a wiser choice.

```suggestion
>(
  state: State<Option, IsMulti, Group>,
  options: Options<Option>,
  alwaysFocusFirstMenuOption: boolean
) {
  const { focusedOption: lastFocusedOption } = state;
  return !alwaysFocusFirstMenuOption &&
    lastFocusedOption &&
    options.indexOf(lastFocusedOption) > -1
    ? lastFocusedOption
    : options[0];
```

Propagating suggestion to next code change.